package/Dockerfile.agent

```
@@ -67,8 +67,10 @@ ENV LOGLEVEL_VERSION v0.1.6
 
 RUN curl -sLf https://github.com/rancher/loglevel/releases/download/${LOGLEVEL_VERSION}/loglevel-${ARCH}-${LOGLEVEL_VERSION}.tar.gz | tar xvzf - -C /usr/bin
 
-LABEL io.cattle.agent true
-ENV AGENT_IMAGE rancher/rancher-agent:${VERSION}
+LABEL io.cattle.agent=true
+
+ARG RANCHER_REPO=rancher
```

```suggestion
ARG RANCHER_REPO=${RANCHER_REPO}
```
Isn't it possible to do something like this and reuse the arg declared on line 2?

---

No. it's not.  In DockerFile, an argument is not shared between layers, so we need to define the arg again here and keep the default value the same as in line 2.

---

I've done something similar with the `VERSION` arg on `package/Dockerfile`, if it isn't possible with this arg i can approve this

---

Hi @tashima42,  I found [this comment](https://github.com/rancher/rancher/commit/ac7e5830c29bbdb3e08632307bd6189d1dceb42a) which seems to be the one you mention. 
In that case,  the VERSION argument is correctly set because the build argument VERSION is defined in the [GHA workflow](https://github.com/rancher/rancher/blob/main/.github/workflows/push-release.yml#L132). It doesn’t derive its value from the environment variable VERSION (whose value is "" in the new stage) 

Docker documentation explains it [here](https://docs.docker.com/reference/dockerfile/#scope):
> An ARG instruction goes out of scope at the end of the build stage where it was defined. To use an argument in multiple stages, each stage must include the ARG instruction.

Please let me know if that makes sense.